File: Prime.java
public class Prime {
    public static boolean isPrime(int n) {
        if (n <= 1) {
            return false;
        }
        if (n <= 3) {
            return true;
        }
        if (n % 2 == 0 || n % 3 == 0) {
            return false;
        }
        int i = 5;
        while (i * i <= n) {
            if (n % i == 0 || n % (i + 2) == 0) {
                return false;
            }
            i += 6;
        }
        return true;
    }

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();

        int[] primes = new int[1000000];
        primes[0] = 2;
        primes[1] = 3;
        int primeCount = 2;

        for (int i = 5; i < 1000000; i++) {
            if (isPrime(i)) {
                primes[primeCount] = i;
                primeCount++;
                System.out.println(i);
            }
        }

        long endTime = System.currentTimeMillis();
        double executionTime = (endTime - startTime) / 1000.0; // Convert to seconds
        System.out.println("Execution Time: " + executionTime + " seconds");
    }
}

File: prime.rb
require 'benchmark'

def is_prime(n)
  return false if n <= 1
  return true if n <= 3
  return false if n % 2 == 0 || n % 3 == 0

  i = 5
  while i * i <= n
    return false if n % i == 0 || n % (i + 2) == 0
    i += 6
  end

  true
end

start_time = Time.now

primes = [2, 3]
(5..999999).each do |i|
  primes << i if is_prime(i)
end

primes.each { |prime| puts prime }

end_time = Time.now
execution_time = end_time - start_time
puts "Execution Time: #{execution_time} seconds"
